src/components/table/table.tsx

```diff
@@ -0,0 +1,106 @@
+import React, { ComponentPropsWithoutRef, forwardRef } from 'react';
+import cn from 'classnames';
+
+import './styles.scss';
+
+export type TableProps = ComponentPropsWithoutRef<'table'>;
+export const Table = forwardRef<HTMLTableElement, TableProps>(function Thead(
+  { className, ...props },
+  ref
+) {
+  return (
+    <table
+      ref={ref}
+      className={cn('govuk-table', 'lbh-table', className)}
+      {...props}
+    />
+  );
+});
+
+export type TheadProps = ComponentPropsWithoutRef<'thead'>;
+export const Thead = forwardRef<HTMLTableSectionElement, TheadProps>(
+  function Thead({ children, ...props }, ref) {
+    return (
+      <thead ref={ref} className="govuk-table__head" {...props}>
+        {children}
+      </thead>
+    );
+  }
+);
+
+export type TbodyProps = ComponentPropsWithoutRef<'tbody'>;
+export const Tbody = forwardRef<HTMLTableSectionElement, TbodyProps>(
+  function Tbody({ children, ...props }, ref) {
+    return (
+      <tbody ref={ref} className="govuk-table__body" {...props}>
+        {children}
+      </tbody>
+    );
+  }
+);
+
+export type TrProps = ComponentPropsWithoutRef<'tr'>;
+export const Tr = forwardRef<HTMLTableRowElement, TrProps>(function Tr(
+  { children, ...props },
+  ref
+) {
+  return (
+    <tr ref={ref} className="govuk-table__row" {...props}>
+      {children}
+    </tr>
+  );
+});
+
+export interface ThProps extends ComponentPropsWithoutRef<'th'> {
+  isNumeric?: boolean;
+}
+export const Th = forwardRef<HTMLTableHeaderCellElement, ThProps>(function Th(
+  { children, isNumeric, ...props },
+  ref
+) {
+  const linkedClasses = cn('govuk-table__cell', 'govuk-table__header', {
+    'govuk-table__cell--numeric': isNumeric,
+  });
+
+  return (
+    <th ref={ref} className={linkedClasses} {...props}>
+      {children}
+    </th>
+  );
+});
+
+export interface TdProps extends ComponentPropsWithoutRef<'td'> {
+  isNumeric?: boolean;
+}
+export const Td = forwardRef<HTMLTableCellElement, TdProps>(function Td(
+  { children, isNumeric },
+  ref
+) {
+  const linkedClasses = cn('govuk-table__cell', {
+    'govuk-table__cell--numeric': isNumeric,
+  });
+
+  return (
+    <td ref={ref} className={linkedClasses}>
+      {children}
+    </td>
+  );
+});
+
+export interface TableCaptionProps extends ComponentPropsWithoutRef<'caption'> {
+  title: string;
```

Comment:
I think instead of title, use children here. I'd check if caption only supports string as a children type as well.

Reply:
https://html.spec.whatwg.org/multipage/tables.html#the-caption-element looks like it supports flow content.